### PR TITLE
Feature/weets api integration

### PR DESCRIPTION
**Pull request by DanielAugustoUruena**
## Summary

- Geek jokes service added
- Endpoint to post a weet created

## Known Issues

None.

## Trello Card

[Servicios / Promises - Witter](https://trello.com/c/gjawXa5L/32-servicios-promises-witter)

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -44,6 +44,9 @@ const config = {
       apiDate: process.env.API_DATE || 'X-API-Date',
       packageVersion: process.env.PACKAGE_VERSION || 'X-Package-Version',
       nodeVersion: process.env.NODE_VERSION || 'X-Node-Version'
+    },
+    geek_joke_api: {
+      url: 'https://geek-jokes.sameerkumar.website/api'
```
Comment (jonnyalexbh) [inline config/index.js]: here we can use an environment variable, what do you think ?

File: app/controllers/weets.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const geekJokeService = require('../services/geek_joke.service');
```
Comment (jonnyalexbh) [inline app/controllers/weets.js]: controller is not required according to acceptance criteria

File: app/routes.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
 // const controller = require('./controllers/controller');
 const { healthCheck } = require('./controllers/healthCheck');
+const weets = require('./controllers/weets');
 
 exports.init = app => {
   app.get('/health', healthCheck);
+  app.post('/weet', weets.postNewWeet);
```
Comment (jonnyalexbh) [inline app/routes.js]: a path is not necessary because it must be tested by console

File: app/services/geek_joke.service.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const axios = require('axios');
+const config = require('../../config');
+
+const getRandomJoke = () => {
+  const options = {
+    params: {
+      format: 'json'
+    }
+  };
+  return axios.default.get(config.common.geek_joke_api.url, options);
```
Comment (jonnyalexbh) [inline app/services/geek_joke.service.js]: when modifying this, remember to add informational logs

File: app/services/geek_joke.service.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const axios = require('axios');
```
Comment (jonnyalexbh) [inline app/services/geek_joke.service.js]: remember that the file name should be plural `weets.js`

**Review by jonnyalexbh — COMMENTED** [review]
Please validate the comments

File: config/index.js (hunk 1 above)
Comment (DanielAugustoUruena) [reply config/index.js]: Agree, I had been forget it.

File: app/controllers/weets.js (hunk 2 above)
Comment (DanielAugustoUruena) [reply app/controllers/weets.js]: Controller removed

File: app/routes.js (hunk 3 above)
Comment (DanielAugustoUruena) [reply app/routes.js]: path removed

File: app/services/geek_joke.service.js (hunk 5 above)
Comment (DanielAugustoUruena) [reply app/services/geek_joke.service.js]: file renamed

File: app/services/geek_joke.service.js (hunk 4 above)
Comment (DanielAugustoUruena) [reply app/services/geek_joke.service.js]: `console.info` added before send request to api

File: config/index.js [hunk 6]
```diff
@@ -44,6 +44,9 @@ const config = {
       apiDate: process.env.API_DATE || 'X-API-Date',
       packageVersion: process.env.PACKAGE_VERSION || 'X-Package-Version',
       nodeVersion: process.env.NODE_VERSION || 'X-Node-Version'
+    },
+    geek_joke_api: {
+      url: process.env.GEEK_JOKES_URL || 'https://geek-jokes.sameerkumar.website/api'
```
Comment (jonnyalexbh) [inline config/index.js:49]: great 🥇

File: app/services/geek_jokes.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+const getRandomJoke = async () => {
+  const options = {
+    params: {
+      format: 'json'
+    }
+  };
+  console.info('Getting random joke...');
+  return axios.default.get(config.common.geek_joke_api.url, options)
+    .then(response => response.data);
```
Comment (jonnyalexbh) [inline app/services/geek_jokes.js]: you would need to handle the error `catch`

Comment (jonnyalexbh) [inline app/services/geek_jokes.js]: you should use the custom Wolox logger (app.logger.js) in the catch with the custom error (app/erros.js)

**Review by jonnyalexbh — APPROVED** [review]

File: app/services/geek_jokes.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+const axios = require('axios');
+const config = require('../../config');
+
+const getRandomJoke = async () => {
```
Comment (jonnyalexbh) [inline app/services/geek_jokes.js]: do you think this(async) is necessary?

File: app/services/geek_jokes.js (hunk 7 above)
Comment (DanielAugustoUruena) [reply app/services/geek_jokes.js]: `console.log` changed to `logger.info`

Comment (DanielAugustoUruena) [reply app/services/geek_jokes.js]: catch block added with error handler

File: app/services/geek_jokes.js (hunk 8 above)
Comment (DanielAugustoUruena) [reply app/services/geek_jokes.js]: async word removed

File: app/errors.js [hunk 9]
```diff
@@ -1,3 +1,5 @@
+const { message } = require("prompt");
```
Comment (jonnyalexbh) [inline app/errors.js:1]: iIthink this is not necessary in the next pr please delete

**Review by jonnyalexbh — APPROVED** [review]
Good job 🥇

**Pull request merged**